lib/pay_day_loan/load_worker.ex

```diff
@@ -26,27 +31,50 @@ defmodule PayDayLoan.LoadWorker do
     GenServer.start_link(__MODULE__, [init_state], gen_server_opts)
   end
 
-  @spec init([PayDayLoan.t()]) :: {:ok, PayDayLoan.t()}
+  @spec init([PayDayLoan.t()]) :: {:ok, state}
   def init([pdl]) do
     Process.send_after(self(), :ping, @startup_dwell)
-    {:ok, pdl}
+    {:ok, %{pdl: pdl, load_task_ref: nil}}
+  end
+
+  @spec handle_cast(atom, state) :: {:noreply, state}
+  def handle_cast(:ping, %{pdl: pdl, load_task_ref: nil} = state) do
+    load_task = start_load_task(pdl)
+    {:noreply, %{state | load_task_ref: load_task.ref}}
+  end
+
+  def handle_cast(:ping, %{pdl: _pdl, load_task_ref: ref} = state) when is_reference(ref) do
+    {:noreply, state}
+  end
+
+  @spec handle_info(atom, state) :: {:noreply, state}
```

Comment:
I think you can just `@impl true` here since elixir 1.5 (which I think we're dropping pre-1.5 support, right?)

Reply:
Yes, that's correct. I added a call to `Supervisor.init` which was adde in version 1.5.